IOMC/RandomEngine/src/RandomNumberGeneratorService.cc

```diff
@@ -146,6 +148,25 @@ namespace edm {
                  "configuration file were " << initialSeedSet[0] << " and " << initialSeedSet[1]
               << "\nThis was for the module with label \"" << label << "\".\n";
           }
+        } else if( engineName == std::string("XorShift128Plus") ) {
+          if( initialSeedSet.size() != 4U ) {
+            throw Exception(errors::Configuration)
+              << "Random engines of type \"XorShift128Plus\" require 4 seeds\n"
+              << "be specified with the parameter named \"initialSeedSet\".\n"
+              << "Either \"initialSeedSet\" was not in the configuration\n"
+              << "or its size was not 4 for the module with label \"" << label << "\".\n" ;
+          }
+          if(initialSeedSet[0] > maxSeedXorShift128Plus ||
+             initialSeedSet[1] > maxSeedXorShift128Plus || 
+             initialSeedSet[2] > maxSeedXorShift128Plus ||
+             initialSeedSet[3] > maxSeedXorShift128Plus) {  // They need to fit in a 32 bit integer
+            throw Exception(errors::Configuration)
+              << "The XorShift128Plus seeds should be in the range 0 to " << maxSeedXorShift128Plus << ".\n"
+              << "The seeds passed to the RandomNumberGenerationService from the\n"
+                 "configuration file were " << initialSeedSet[0] << " , " << initialSeedSet[1]
+              << " , " << initialSeedSet[2] << " , and " << initialSeedSet[3]
+              << "\nThis was for the module with label \"" << label << "\".\n";
+          }
```

Comment:
It would probably be worth checking that all four seeds are not zero. Actually, given the offsets that might be applied to the first seed by incrementing it, if you check here you would need to check that the last 3 seeds are not all zero to be safe. And maybe a comment explaining that seed restrictions would be nice.
Also, it does not do much harm, but the check that the seeds are not too large seems unnecessary since they are stored in 32 bit integers and the max value is the max value of a 32 bit integer.